### PR TITLE
Container-based GH actions

### PR DESCRIPTION
**Pull request by ischoegl**
**Changes proposed in this pull request**

- Add CI tests running on [official ubuntu](https://hub.docker.com/_/ubuntu) docker images
- ~Currently added to `main.yaml` to facilitate review; should be moved to `post-merge-tests.yaml` before merging~
- ~Move mingw runner to `post-merge-tests`~
- Remove vestigial `h5py` references

**If applicable, provide an example illustrating new features this pull request is introducing**

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

**Comment by ischoegl** [comment]
~As mentioned on top, the new runner should be moved before merging. It is currently in `main.yaml` so the CI is actually triggered and output can be reviewed.~ *Edit: now moved to final location.*

**Comment by ischoegl** [comment]
I moved the docker tests to `post-merge-tests` to expedite things. Fwiw, the output of the CI run can be reviewed in previously successful GH Action runs, i.e.:
* https://github.com/Cantera/cantera/actions/runs/5994969657/job/16257310506, and
* https://github.com/Cantera/cantera/actions/runs/5994969657/job/16257310399

File: .github/workflows/post-merge-tests.yml [hunk 1]
```diff
@@ -47,3 +47,114 @@ jobs:
     - name: Test Cantera
       run:
         python3 `which scons` test show_long_tests=yes verbose_tests=yes --debug=time
+
+  windows-mingw:
+    name: mingw on Windows, Python 3.10
+    runs-on: windows-2022
+    timeout-minutes: 120 # MinGW is slooooow
```
Comment (speth) [inline .github/workflows/post-merge-tests.yml]: I think I'd prefer to keep the `mingw` run as part of the normal CI. It does occasionally catch some odd issues, and despite the comment here, these CI runs are actually faster than the MSVC ones in many cases.  There must have been some updates to the MinGW packages that helped with compilation time.

**Review by speth — COMMENTED** [review]
The new jobs look good to me. Thanks for cleaning up the bits about h5py that I missed in #1605. My only suggestion is to keep the MinGW runner where it is.

File: .github/workflows/post-merge-tests.yml (hunk 1 above)
Comment (ischoegl) [reply .github/workflows/post-merge-tests.yml]: No problem - dropped that commit.

**Comment by ischoegl** [comment]
@speth ... thanks for the feedback! No problem with leaving mingw where it was. Other than that, I believe we have a decent template to be used for regression tests with other Linux flavors in the future.

**Review by speth — APPROVED** [review]

**Pull request merged**